examples/template_engine.du

```diff
@@ -0,0 +1,58 @@
+import System;
+
+const LEFT_BRACE = "{";
+const RIGHT_BRACE = "}";
+
+// Template is a simple template rendering engine.
+class Template {
+    private tmpl;
+    private klass;
+    private rendered;
+
+    init(tmpl, klass) {
+        this.tmpl = tmpl;
+        this.klass = klass;
+    }
```

Comment:
Just some suggestions here, open for discussion.

- I've moved the `LEFT_BRACE` / `RIGHT_BRACE` constants inside the class to be class variables rather than constants outside of the class
- I've removed the `rendered` private property. My rationale for this is it's only ever used inside render and is defined as private. If it was public it could potentially have use for caching, but as it's private it doesn't need to be on the object.
- Ive removed the explicit private / initialisation for tmpl and klass for their implicit property counterparts.

```suggestion
// Template is a simple template rendering engine.
class Template {
    var LEFT_BRACE = "{";
    var RIGHT_BRACE = "}";

    init(private tmpl, private klass) {}
```

Reply:
All of that seems fine. Makes me wonder how hard it would be to implement constants on classes. Or even a const key word in function args.

Reply:
I think constants on classes shouldn't be _too_ bad. It would pretty much be the same flow as class vars currently but potentially to a new internal Table within the class.

e.g: https://github.com/dictu-lang/Dictu/blob/develop/src/vm/object.h#L223

We could add
```c
Table constantProperties;
```

https://github.com/dictu-lang/Dictu/blob/develop/src/vm/vm.c#L1254-L1260

Here we could extend the opcode where `var` would add to `publicProperties` and `const` would add to `constantProperties`.

https://github.com/dictu-lang/Dictu/blob/develop/src/vm/vm.c#L1014-L1022

The following bit of code would need to be extended to also check for constants (in all places it occurs)

```c
while (klass != NULL) {
                        if (tableGet(&klass->publicProperties, name, &value)) {
                            pop(vm); // Instance.
                            push(vm, value);
                            DISPATCH();
                        }

                        if (tableGet(&klass->constantProperties, name, &value)) {
                            pop(vm); // Instance.
                            push(vm, value);
                            DISPATCH();
                        }

                        klass = klass->superclass;
                    }
```

Finally here: https://github.com/dictu-lang/Dictu/blob/develop/src/vm/vm.c#L1229-L1236

We would probably need to check if the property attempting to be assigned to the class isn't already present in the constants table